src/lib/table/table.spec.ts

```diff
@@ -1,52 +1,211 @@
+      // Add data
+      component.underlyingDataSource.addData();
+      fixture.detectChanges();
+      expectTableToMatchContent(tableElement, [
+        ['Column A', 'Column B', 'Column C'],
+        ['a_1', 'b_1', 'c_1'],
+        ['a_2', 'b_2', 'c_2'],
+        ['a_3', 'b_3', 'c_3'],
+        ['a_4', 'b_4', 'c_4'],
+      ]);
+
+      // Remove data
+      const modifiedData = dataSource.data.slice();
+      modifiedData.shift();
+      dataSource.data = modifiedData;
+      fixture.detectChanges();
+      expectTableToMatchContent(tableElement, [
+        ['Column A', 'Column B', 'Column C'],
+        ['a_2', 'b_2', 'c_2'],
+        ['a_3', 'b_3', 'c_3'],
+        ['a_4', 'b_4', 'c_4'],
+      ]);
+    });
+
+    it('should be able to filter the table contents', fakeAsync(() => {
+      // Change filter to a_1, should match one row
+      dataSource.filter = 'a_1';
+      fixture.detectChanges();
+      expectTableToMatchContent(tableElement, [
+        ['Column A', 'Column B', 'Column C'],
+        ['a_1', 'b_1', 'c_1'],
+      ]);
+      flushMicrotasks();  // Resolve promise that updates paginator's length
+      expect(dataSource.paginator!.length).toBe(1);
+
+      // Change filter to a_2, should match one row
+      dataSource.filter = 'a_2';
+      fixture.detectChanges();
+      expectTableToMatchContent(tableElement, [
+        ['Column A', 'Column B', 'Column C'],
+        ['a_2', 'b_2', 'c_2'],
+      ]);
+
+      // Change filter to empty string, should match all rows
+      dataSource.filter = '';
+      fixture.detectChanges();
+      expectTableToMatchContent(tableElement, [
+        ['Column A', 'Column B', 'Column C'],
+        ['a_1', 'b_1', 'c_1'],
+        ['a_2', 'b_2', 'c_2'],
+        ['a_3', 'b_3', 'c_3'],
+      ]);
+
+      // Change filter function and filter, should match to rows.
+      dataSource.filterTermAccessor = data => {
+        switch (data.a) {
+          case 'a_1': return 'elephant';
+          case 'a_2': return 'zebra';
+          case 'a_3': return 'monkey';
+          default: return '';
+        }
+      };
+      dataSource.filter = 'zebra';
+      fixture.detectChanges();
+      expectTableToMatchContent(tableElement, [
+        ['Column A', 'Column B', 'Column C'],
+        ['a_2', 'b_2', 'c_2'],
+      ]);
+    }));
+
+    it ('should be able to sort the table contents', () => {
+      // Activate column A sort
+      component.sort.sort(component.sortHeader);
+      fixture.detectChanges();
+      expectTableToMatchContent(tableElement, [
+        ['Column ASorted by a ascending', 'Column B', 'Column C'],
+        ['a_1', 'b_1', 'c_1'],
+        ['a_2', 'b_2', 'c_2'],
+        ['a_3', 'b_3', 'c_3'],
+      ]);
+
+      // Activate column A sort again (reverse direction)
+      component.sort.sort(component.sortHeader);
+      fixture.detectChanges();
+      expectTableToMatchContent(tableElement, [
+        ['Column ASorted by a descending', 'Column B', 'Column C'],
```

Comment:
Why is there no space between `Column A` and `Sorted`?

Reply:
Side-effect of how I set up the `expectTableToMatchContent` - it converts a cell to its `.textContent`, which includes visually hidden elements. Sorted headers have a `cdk-visually-hidden` element that describes the sort.

In our case here, we've got three siblings in the cell (approx.): 
```
<button> Column A </button> 
<div class="sort-arrow"></div> 
<span class="cdk-visually-hidden"> Sorted by a descending </span>
```

This converts to `Column ASorted by a descending`

Reply:
This will be an issue when we turn on whitespace collapsing; we should add an `&nbsp;` between those

Reply:
SGTM - added an HTML space before the description label and changed the tests to check for that break